shared/zodSchemas.ts

```diff
@@ -356,3 +353,62 @@ export const CursorPaginationDatetimeSchema = CursorPaginationSchema.extend({
 export type CursorPaginationDatetimeParams = z.infer<
   typeof CursorPaginationDatetimeSchema
 >;
+
+///////////////////////////////
+// INSTITUTION SCHEMAS
+///////////////////////////////
+export const createInstitutionSchema = z.object({
+  institutionName: z.string(),
+  institutionDomain: z.string(),
+});
+
+export const institutionNameSchema = z
+  .string()
+  .refine((data) => data.length > 0, {
+    message: "Institution name cannot be null",
+  });
+
+export const institutionDomainSchema = z
+  .string()
+  .refine((data) => data.length > 0, {
+    message: "Institution domain cannot be null",
+  });
+
+export const institutionIDSchema = z
+  .string()
+  .uuid()
+  .refine((data) => data.length > 0, {
+    message: "ID is invalid",
+  });
+
+///////////////////////////////
+// USER SCHEMAS
+///////////////////////////////
+export const createUserSchema = z.object({
```

Comment:
I think we should define createUserSchema based off the UserSchema we defined earlier

Reply:
This will probably work

```

export const UserSchema = z.object({
  id: z.string().uuid(),
  username: z.string().min(3).max(20),
  firstName: z.string().min(3).max(20),
  lastName: z.string().min(3).max(20),
  email: z.string().email({ message: "Invalid email address" }).min(5),
  password: z
    .string()
    .min(8, { message: "Password must be greater than 8 characters long" }),
  yearOfStudy: z.coerce
    .number()
    .min(1, { message: "Year of Study must be greater than 0" })
    .max(10, { message: "Year of Study must be less than 11" }),
  institutionId :z.string().uuid(),
  isVerified: BooleanSchema,
  profilePic: z.string().nullable(),
  otp: z.string(),
  jwt: z.string(),
});


export const createUserSchema = UserSchema.omit({
  id: true,
  isVerified: true,
  otp: true,
  jwt: true,
});
```

I think we should also add a createdAt field to the user model, so we know when the user signed up. 
Maybe in a future pr, because we would probably have to adjust the seed script too
```createdAt      DateTime            @unique @default(now()) @map("created_at")```